recipes/openimageio/all/conanfile.py

```diff
@@ -63,260 +64,250 @@ class OpenImageIOConan(ConanFile):
-        cmake.definitions["USE_FIELD3D"] = False
-        cmake.definitions["USE_GIF"] = self.options.with_giflib
-        cmake.definitions["USE_LIBHEIF"] = self.options.with_libheif
-        cmake.definitions["USE_LIBRAW"] = self.options.with_raw
-        cmake.definitions["USE_OPENVDB"] = self.options.with_openvdb
-        cmake.definitions["USE_PTEX"] = self.options.with_ptex
-        cmake.definitions["USE_R3DSDK"] = False
-        cmake.definitions["USE_NUKE"] = False
-        cmake.definitions["USE_OPENGL"] = False
-        cmake.definitions["USE_QT"] = False
-        cmake.definitions["USE_LIBPNG"] = self.options.with_libpng
-        cmake.definitions["USE_FREETYPE"] = self.options.with_freetype
-        cmake.definitions["USE_LIBWEBP"] = self.options.with_libwebp
-        cmake.definitions["USE_OPENJPEG"] = self.options.with_openjpeg
+        tc.variables["USE_HDF5"] = self.options.with_hdf5
+        tc.variables["USE_OPENCOLORIO"] = self.options.with_opencolorio
+        tc.variables["USE_OPENCV"] = self.options.with_opencv
+        tc.variables["USE_TBB"] = self.options.with_tbb
+        tc.variables["USE_DCMTK"] = self.options.with_dicom
+        tc.variables["USE_FFMPEG"] = self.options.with_ffmpeg
+        tc.variables["USE_FIELD3D"] = False
+        tc.variables["USE_GIF"] = self.options.with_giflib
+        tc.variables["USE_LIBHEIF"] = self.options.with_libheif
+        tc.variables["USE_LIBRAW"] = self.options.with_raw
+        tc.variables["USE_OPENVDB"] = self.options.with_openvdb
+        tc.variables["USE_PTEX"] = self.options.with_ptex
+        tc.variables["USE_R3DSDK"] = False
+        tc.variables["USE_NUKE"] = False
+        tc.variables["USE_OPENGL"] = False
+        tc.variables["USE_QT"] = False
+        tc.variables["USE_LIBPNG"] = self.options.with_libpng
+        tc.variables["USE_FREETYPE"] = self.options.with_freetype
+        tc.variables["USE_LIBWEBP"] = self.options.with_libwebp
+        tc.variables["USE_OPENJPEG"] = self.options.with_openjpeg
 
-        if self.options.with_openvdb:
-            cmake.definitions["CMAKE_CXX_STANDARD"] = 14
-
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+        tc.generate()
+        cd = CMakeDeps(self)
+        cd.generate()
 
     def build(self):
-        self._patch_sources()
-
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE.md", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
```

Comment:
In https://c3i.jfrog.io/c3i/misc/logs/pr/19950/38-windows-visual_studio/openimageio/2.3.7.2//b2c07836b65cee5d00045084ae8eafc572e6d078-build.txt i see an error about system dlls in bien folder.

Reply:
@irieger Try something like this: https://github.com/conan-io/conan-center-index/blob/a523d487eeb89c3c5cdef54d4a8d15a6d1df4974/recipes/mongo-cxx-driver/all/conanfile.py#L155-L160

Reply:
Yeah, saw that message too. Have nearly no clue about the build on crap OS. However your rmdir seems a bit wide reaching to me without knowing how exactly the folders on Windows might look here. Would hope that pkgconfig and cmake aren't filled with actual dlls? Did you test that on Windows?

Reply:
The rmdir isn't necessarilly needed, and your library may not contain them to begin with. I was just showing context (lines above and below).